flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java

```diff
@@ -265,11 +281,15 @@ public String toString() {
 	// ------------------------------------------------------------------------
 
 	private void returnMemorySegment(MemorySegment segment) {
+		assert Thread.holdsLock(availableMemorySegments);
```

Comment:
Hi, I have a question about assert, because I found that assertion is disabled in java by default. why not use explicit `synchronized(availableMemorySegments)` which may be more common usage.